node/node.go

```diff
@@ -676,27 +677,11 @@ func (n *Node) checkSenderIsInShard(tx *transaction.Transaction) error {
 }
 
 // CreateTransaction will return a transaction from all the required fields
-func (n *Node) CreateTransaction(
-	nonce uint64,
-	value string,
-	receiver string,
-	receiverUsername []byte,
-	sender string,
-	senderUsername []byte,
-	gasPrice uint64,
-	gasLimit uint64,
-	dataField []byte,
-	signatureHex string,
-	chainID string,
-	version uint32,
-	options uint32,
-	guardian string,
-	guardianSigHex string,
-) (*transaction.Transaction, []byte, error) {
-	if version == 0 {
+func (n *Node) CreateTransaction(txArgs *external.ArgsCreateTransaction) (*transaction.Transaction, []byte, error) {
```

Comment:
can you also check for nil txArgs?
so we don't get a panic on dereferencing it below

Reply:
added nil args check